### PR TITLE
[FUCKED UP] Bobnames: The Masquerade

### PR DESCRIPTION
**Pull request by ChungusGamer666**
## About The Pull Request

Essentially, this is a complete rework of the SS13 "anonymity" system.
Now, knowing people's names is handled by a guestbook datum. If they are not in your guestbook, you don't know their names - simple as. Therefore they just appear as something like "young adult man" or "elderly woman" when speaking, or when looking at them.
If you actually KNOW someone and their face/voice isn't hidden, their guestbook name will be shown on examine.

## Why It's Good For The Game

Actually insane roleplay potential, prevents imbeciles like me from slipping up and saying the name of someone they first met 3 seconds ago. Now you will have to put effort in to memorize anyone who isn't in your faction.

## Changelog

:cl:
refactor: Bobnames is here. Woe be upon ye.
/:cl:

### CODE REVIEW

**Comment by Hekzder** [comment]
wow!

**Comment by ChungusGamer666** [comment]
ready for review 2 electric boogaloo big chungu

**Comment by ProfessorPopoff** [comment]
Nice linters....

File: code/datums/guestbook.dm [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+	var/mob/living/living_user = user
+	if(istype(living_user) && !living_user.in_fov(guest))
+		if(!silent)
+			to_chat(user, span_warning("You can't see them!"))
+		return FALSE
+	var/visible_name = guest.get_visible_name("")
+	var/face_name = guest.get_face_name("")
+	if(!visible_name || !face_name)
+		if(!silent)
+			to_chat(user, span_warning("You can't see their very well!"))
```
Comment (ProfessorPopoff) [inline code/datums/guestbook.dm]: You can't see their very well!

Comment (ChungusGamer666) [reply code/datums/guestbook.dm:147]: soul

**Comment by ProfessorPopoff** [comment]
This seems pretty good. I can't get it to break and the only way to test further is to have it in for the tests. @Koshenko @Hekzder figure it out :)

**Comment by ChungusGamer666** [comment]
this server needs a hero
gassywwe

**Pull request merged**